openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -53,40 +54,54 @@
  * @since 2.1.0
  */
 public class OASModifier {
-    OpenAPI openapi;
     List<Diagnostic> diagnostics = new ArrayList<>();
 
-    public OASModifier(OpenAPI openapi) {
-        this.openapi = openapi;
-    }
 
     public List<Diagnostic> getDiagnostics() {
         return diagnostics;
     }
 
-    //modifyWithBallerinaConventions 
-    public OpenAPI modifyWithBallerinaConventions() {
+    public Optional<Map<String, String>> getProposedNameList(OpenAPI openapi) {
```

Comment:
Does this only proposed names for the types? In that case how to we handle the (path) parameter names?

Reply:
This is for type, I prefer to have that `getResolvedNameMap` function for handling duplicates here https://github.com/ballerina-platform/openapi-tools/blob/ee134662d9ab9bebbb56051181a6c3b381e45eea/openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java#L338 if possible WDYT?